### ISSUE TITLE
Database error: "Prepared statement needs to be re-prepared"

### ISSUE THREAD

**Issue by 4drian3d** [A248/LibertyBans#36]
In a new installation of the plugin in Velocity, connected by MariaDB gives the following error when users enter https://pastebin.com/AUMRPpGH .
When installing the plugin for the first time, it works "correctly" https://pastebin.com/CVU6u3Xf

**Comment by A248** [comment]
What version of MariaDB are you using? This error does not happen when testing on MariaDB 10.3 or 10.5.

Many sources suggest this is a long-standing bug in MariaDB, which *may* be fixed in later MariaDB versions and does have a moderately effective workaround. It is related to prepared statements and table views.

The immediate workaround is to increase the [`table_definition_cache`](https://mariadb.com/kb/en/server-system-variables/#table_definition_cache) variable, from the default of 1024, to a value like 16384, to alleviate the issue. https://stackoverflow.com/a/4381155

Both [MySQL](https://bugs.mysql.com/bug.php?id=42041) and [MariaDB](https://jira.mariadb.org/browse/MDEV-17124) have open issues for this bug in their tracker. Also, this commit in MariaDB 10.1.34 suggests it may be fixed in later versions: https://github.com/MariaDB/server/commit/5e63f616a7ed511e6e69581f6659f60f1ec8633b

Additionally, this bug is partially induced by increased load, which is why others have been able to reproduce it using `mysqldump`. If you have other plugins which are not properly closing their prepared statements, . Those plugins will cause you memory leaks on your server or proxy and they will cause this error on MariaDB by filling up the table cache.

**Comment by 4drian3d** [comment]
I don't have any other plugin that uses MariaDB or MySQL in Velocity, LibertyBans is the first one that I put that uses external database. And from what I see, the version that uses the host where I have the database: 10.3.21-MariaDB-log - MariaDB Server. Can't there be another momentary solution besides changing the table_definition_cache value? I don't have the necessary permissions in the host where the database is.

**Comment by A248** [comment]
Okay, good.

> 10.05 14:41:56 [Server] ERROR [space.arim.libertybans.core.env.ParallelisedListener]: Critical: Event PlayerChatEvent{player=[connected player] ------ (/------:----), message=los plots? :v, result=allowed} was previously blocked by the server or another plugin, but since then, some plugin has uncancelled the blocking. This may lead to bans or mutes not being checked and enforced.

This message is related to a different topic. I've explained it in this wiki page:
https://github.com/A248/LibertyBans/wiki/Event-was-previously-blocked-by-the-server-or-another-plugin...

In short, this is supposed to be a warning that another plugin has undone a cancelled chat event. Un-cancelling a chat event can lead to mutes not being enforced, because of this misbehaving plugin.

If you have no other plugins present on the same server where LibertyBans is running, please open a new issue about this message. I will need more information about your environment (platform, LibertyBans version, platform version) if you do so.

**Comment by 4drian3d** [comment]
At the moment of restarting Velocity completely (end and start) the error seems to have been corrected.

Just in case...

11.05 10:04:03 [Server] INFO Plugins: authmevelocity, commandwhitelist, epicguard, libertybans, luckperms, message, minimotd-velocity, nuvotifier, skinsrestorer
11.05 10:04:03 ------@gmail.com ran command: velocity plugins
11.05 10:07:00 [Server] INFO Velocity 1.1.6-SNAPSHOT (git-06ecab26-b430)
11.05 10:07:00 [Server] INFO Copyright 2018-2021 Velocity Contributors. Velocity is licensed under the terms of the GNU General Public License v3.
11.05 10:07:00 [Server] INFO Visit the Velocity website or the Velocity GitHub
11.05 10:07:00 -------@gmail.com ran command: velocity version
Java Version: 11.0.8

**Comment by A248** [comment]
Closing in favor of [MDEV-17124](https://jira.mariadb.org/browse/MDEV-17124)
